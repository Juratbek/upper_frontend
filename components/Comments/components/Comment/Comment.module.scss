@import 'styles/_mixins';

.comment {
  padding: 0.5rem;
  position: relative;

  & .author {
    display: flex;
  }
  & .date {
    font-size: 0.7rem;
  }

  & .message {
    font-size: 0.8rem;
    margin-top: 0.3rem;
    overflow-wrap: break-word;
  }

  &:hover .edit-button {
    display: block;
  }
}

.edit-button {
  position: absolute;
  display: none;
  top: 5px;
  right: 0;

  @include screenMax576 {
    display: block;
    color: gray;
  }
}
