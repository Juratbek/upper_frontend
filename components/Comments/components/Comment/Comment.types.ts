import { IComment } from 'types';

export interface ICommentProps extends IComment {
  onEditClick: (comment: IComment) => void;
}
