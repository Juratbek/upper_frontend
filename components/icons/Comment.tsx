import { IIconProps } from './types';

export const CommentIcon = ({
  width = 24,
  height = 24,
  color = 'black',
  variant = 'outlined',
}: IIconProps) => (
  <svg width={width} height={height} viewBox='0 0 24 24' fill='none'>
    {variant === 'fulfilled' && (
      <path
        d='M2 19.7307V6.05541C2 4.82811 2.99492 3.83319 4.22222 3.83319H19.7778C21.0051 3.83319 22 4.82811 22 6.05541V15.51C22 16.7373 21.0051 17.7322 19.7778 17.7322H7.5125C6.83742 17.7322 5.77723 18.5662 5.77723 18.5662L3.5 19.9999C3.5 19.9999 2 20.5163 2 19.7307Z'
        fill='#007AFF'
      />
    )}
    {variant === 'outlined' && (
      <path
        d='M2 19.6392V5.96392C2 4.73662 2.99492 3.7417 4.22222 3.7417H19.7778C21.0051 3.7417 22 4.73662 22 5.96392V15.4185C22 16.6458 21.0051 17.6407 19.7778 17.6407H7.5125C6.83742 17.6407 5.77723 18.4747 5.77723 18.4747L3.18724 20.0556C3.18724 20.0556 2 20.7213 2 19.6392Z'
        stroke={color}
        strokeWidth='1.4'
      />
    )}
    <path d='M6.5 13.0915H17.5' stroke={color} strokeLinecap='round' strokeLinejoin='round' />
    <path
      d='M6.50101 9.09153L15.501 9.09149'
      stroke={color}
      strokeLinecap='round'
      strokeLinejoin='round'
    />
  </svg>
);
