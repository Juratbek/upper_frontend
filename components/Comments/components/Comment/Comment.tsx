import { Avatar } from 'components';
import { Divider } from 'components/Divider/Divider';
import Link from 'next/link';
import { FC } from 'react';
import { toDateString } from 'utils';

import classes from './Comment.module.scss';
import { TCommentProps } from './Comment.types';

export const Comment: FC<TCommentProps> = ({ author, date, text }) => {
  return (
    <>
      <div className={classes.comment}>
        <div className={classes.author}>
          <Avatar imgUrl={author.imgUrl} size='small' />
          <div className='ms-1'>
            <Link href={`/blogs/${author.id}`}>
              <h5 className='m-0 link pointer'>{author.name}</h5>
            </Link>
            <p className={`m-0 ${classes.date}`}>{toDateString(date)}</p>
          </div>
        </div>
        <div className={classes.message}>{text}</div>
      </div>
      <Divider className='w-90 mx-auto' />
    </>
  );
};
