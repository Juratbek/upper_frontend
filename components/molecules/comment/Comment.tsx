import { Avatar, Link } from 'components/lib';
import { FC } from 'react';
import { addAmazonUri } from 'utils/blog';
import { dateInterval } from 'utils/date';

import classes from './Comment.module.scss';
import { ICommentProps } from './Comment.types';

export const Comment: FC<ICommentProps> = (props) => {
  const { author, text, date, updatedText } = props;

  return (
    <div className={classes.root}>
      <Avatar imgUrl={addAmazonUri(author).imgUrl} size='small' />
      <div className={classes.body}>
        <div className={classes['body-header']}>
          <Link href={`/blogs/${author.id}`} className={classes.link}>
            <h4 className={classes['author-name']}>{author.name}</h4>
          </Link>
          <p className={classes.date}>{dateInterval(date)}</p>
        </div>
        <p className={classes.text}>{updatedText || text}</p>
      </div>
    </div>
  );
};
