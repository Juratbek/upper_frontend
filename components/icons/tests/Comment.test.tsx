import { render, screen } from '@testing-library/react';

import { CommentIcon } from '../Comment';

describe('Comment icon', () => {
  it('renders the icon', () => {
    render(<CommentIcon />);
    const icon = screen.getByRole('icon');
    expect(icon).toMatchSnapshot();
  });
});
