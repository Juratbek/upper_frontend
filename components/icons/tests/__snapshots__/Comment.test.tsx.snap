// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Comment icon > renders the icon 1`] = `
<svg
  fill="none"
  height="24"
  role="icon"
  viewBox="0 0 24 24"
  width="24"
>
  <path
    d="M2 19.6392V5.96392C2 4.73662 2.99492 3.7417 4.22222 3.7417H19.7778C21.0051 3.7417 22 4.73662 22 5.96392V15.4185C22 16.6458 21.0051 17.6407 19.7778 17.6407H7.5125C6.83742 17.6407 5.77723 18.4747 5.77723 18.4747L3.18724 20.0556C3.18724 20.0556 2 20.7213 2 19.6392Z"
    stroke="black"
    stroke-width="1.4"
  />
  <path
    d="M6.5 13.0915H17.5"
    stroke="black"
    stroke-linecap="round"
    stroke-linejoin="round"
  />
  <path
    d="M6.50101 9.09153L15.501 9.09149"
    stroke="black"
    stroke-linecap="round"
    stroke-linejoin="round"
  />
</svg>
`;
